### PR TITLE
a simplistic Array.multiNewArray implementation

### PR DESCRIPTION
**Pull request by dgrove-oss**

### CODE REVIEW

**Comment by dgrove-oss** [comment]
Since we don't have Array.set yet, this hasn't actually been tested...mainly submitting to get opinions on whether we want to use Array.set or if we should add an intrinsic for an unchecked store into a referenceArray for qbicc-0.3.0.

**Comment by dgrove-oss** [comment]
Ignore above comment.  Had the obvious "insight" overnight that all the spine arrays must be subclasses of Object[], so we can just cast and do a normal array assignment.  Apparently should have stopped coding a little earlier last night...

**Comment by dgrove-oss** [comment]
This is now ready for review.  It turned out to be an interesting trip down the rabbit hole....it exposed both a bug in our array store check implementation (https://github.com/qbicc/qbicc/pull/934) and that we are not actually initializing the `id` field for the `Class` instances that correspond to primitive types (working on fixing that....).

File: java.base/src/main/java/java/lang/reflect/Array$_native.java [hunk 1]
```diff
@@ -38,4 +38,36 @@ private static Object newArray(Class<?> componentType, int length) throws Negati
...
+
+    private static Object multiNewArray(Class<?> componentType, int[] dimensions) throws IllegalArgumentException, NegativeArraySizeException {
+        int dimCount = dimensions.length;
+        if (componentType.isPrimitive()) {
+            dimCount -= 1;
+        } else {
+            while (componentType.isArray()) {
+                dimCount += 1;
+                componentType = componentType.getComponentType();
+            }
```
Comment (dmlloyd) [inline java.base/src/main/java/java/lang/reflect/Array$_native.java:50]: This is not something we have to fix now, but it should be possible to get the dimensions and leaf type directly from any reference array class thus avoiding a loop.

**Review by dmlloyd — APPROVED** [review]

**Pull request merged**